packages/components/src/slot-fill/bubbles-virtually/use-slot-fills.js

```diff
@@ -0,0 +1,21 @@
+// @ts-nocheck
+/**
+ * External dependencies
+ */
+import { useSnapshot } from 'valtio';
+
+/**
+ * WordPress dependencies
+ */
+import { useContext } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import SlotFillContext from './slot-fill-context';
+
+export default function useSlotFills( name ) {
+	const registry = useContext( SlotFillContext );
+	const fills = useSnapshot( registry.fills );
+	return fills.get( name );
```

Comment:
The important bit here is that, this hook only causes a re-render if the "fills" of a given slot name change change, not any fills.